File: gems/decomplex/examples/kotlin/implicit-control-flow.kt
class FlowExample {
  var status = 0
  var valid = false
  var done = false
  fun prepare() { this.status = 1 }
  fun validate() { this.valid = this.status == 1 }
  fun commit() { this.done = this.valid }
  fun ok1() { this.prepare(); this.validate(); this.commit() }
  fun ok2() { this.prepare(); this.validate(); this.commit() }
  fun ok3() { this.prepare(); this.validate(); this.commit() }
  fun ok4() { this.prepare(); this.validate(); this.commit() }
  fun drift() { this.validate(); this.prepare(); this.commit() }
}

File: gems/decomplex/examples/lua/implicit-control-flow.lua
FlowExample = {}
function FlowExample:prepare() self.status = 1 end
function FlowExample:validate() self.valid = self.status == 1 end
function FlowExample:commit() self.done = self.valid end
function FlowExample:ok1() self.prepare(); self.validate(); self.commit() end
function FlowExample:ok2() self.prepare(); self.validate(); self.commit() end
function FlowExample:ok3() self.prepare(); self.validate(); self.commit() end
function FlowExample:ok4() self.prepare(); self.validate(); self.commit() end
function FlowExample:drift() self.validate(); self.prepare(); self.commit() end
